src/wallet/wallet.cpp

```diff
@@ -1722,8 +1722,12 @@ int CWallet::ScanForWalletTransactions(CBlockIndex *pindexStart, bool fUpdate) {
 
         // no need to read and scan block, if block was created before
         // our wallet birthday (as adjusted for block time variability)
-        while (pindex && nTimeFirstKey && (pindex->GetBlockTime() < (nTimeFirstKey - 7200)))
-            pindex = chainActive.Next(pindex);
+        // if you are recovering wallet with mnemonics start rescan from block when mnemonics implemented in Zcoin
+        if (fRecoverMnemonic)
+            pindex = chainActive[chainParams.GetConsensus().nMnemonicBlock];
```

Comment:
I don't like that here there is no check if `nMnemonicBlock` is less than current height. Looking at the code it seems like it will work anyway (`operator []` will return null and then everything is ok if `pindex` is null). But in case of subsequent changes to the code it might lead to unpleasant bug. Can we set `pindex` to chain tip if that's the case?

Reply:
it will be ok if we set chain tip in case of null.